kotsadm/web/src/components/gitops/GitOpsDeploymentManager.jsx

```diff
@@ -113,21 +104,61 @@ class GitOpsDeploymentManager extends React.Component {
     }
   }
 
+  getGitopsState = async () => {
+    const freshGitops = await this.getGitops()
```

Comment:
`getGitops` throws an error on `catch` but it's not caught here. you can wrap this whole function in `try/catch` or move the `setState` functionality to `getGitops` function (which i recommend)

Reply:
i feel like merging `getGitopsState` with `getGitops` would prevent future mistakes of calling the wrong function

Reply:
yeah, initially there was a reason to have those independent
but not really anymore